src/main/java/net/javadiscord/javabot/systems/help/HelpListener.java

```diff
@@ -129,46 +134,58 @@ private boolean isInvalidForumPost(@NotNull Channel channel) {
 	}
 
 	private boolean isInvalidHelpForumChannel(@NotNull ForumChannel forum) {
-		HelpForumConfig config = botConfig.get(forum.getGuild()).getHelpForumConfig();
+		HelpConfig config = botConfig.get(forum.getGuild()).getHelpConfig();
 		return config.getHelpForumChannelId() != forum.getIdLong();
 	}
 
-	private void handleHelpThanksInteraction(@NotNull ButtonInteractionEvent event, @NotNull ForumHelpManager manager, String @NotNull [] id) {
+	private void handleHelpThanksInteraction(@NotNull ButtonInteractionEvent event, @NotNull HelpManager manager, String @NotNull [] id) {
 		ThreadChannel post = manager.getPostThread();
 		HelpConfig config = botConfig.get(event.getGuild()).getHelpConfig();
 		if (event.getUser().getIdLong() != post.getOwnerIdLong()) {
 			Responses.warning(event, "Sorry, only the person who reserved this channel can thank users.").queue();
 			return;
 		}
 		switch (id[2]) {
-			case "done" -> {
-				List<Button> buttons = event.getMessage().getButtons();
-				// immediately delete the message
-				event.getMessage().delete().queue(s -> {
-					// close post
-					manager.close(event, false, null);
-					// add experience
-					try {
-						HelpExperienceService service = new HelpExperienceService(dataSource, botConfig, helpAccountRepository, helpTransactionRepository);
-						Map<Long, Double> experience = HelpChannelManager.calculateExperience(HELP_POST_MESSAGES.get(post.getIdLong()), post.getOwnerIdLong(), config);
-						for (Map.Entry<Long, Double> entry : experience.entrySet()) {
-							service.performTransaction(entry.getKey(), entry.getValue(), HelpTransactionMessage.HELPED, config.getGuild());
-						}
-					} catch (DataAccessException e) {
-						ExceptionLogger.capture(e, getClass().getName());
-					}
-					// thank all helpers
-					buttons.stream().filter(ActionComponent::isDisabled)
-							.filter(b -> b.getId() != null)
-							.forEach(b -> manager.thankHelper(event, post, Long.parseLong(ComponentIdBuilder.split(b.getId())[2])));
-				});
-
-			}
+			case "done" -> handleThanksCloseButton(event, manager, post, config);
 			case "cancel" -> event.getMessage().delete().queue();
-			default -> event.editButton(event.getButton().asDisabled()).queue();
+			default -> {
+				List<Button> thankButtons = event.getMessage().getButtons().stream()
+						.filter(b -> b.getId() != null &&
+								!ComponentIdBuilder.split(b.getId())[2].equals("done") &&
+								!ComponentIdBuilder.split(b.getId())[2].equals("cancel")
+						).toList();
+				if (thankButtons.stream().filter(Button::isDisabled).count() == thankButtons.size() - 1) {
+					handleThanksCloseButton(event, manager, post, config);
+				} else {
+					event.editButton(event.getButton().asDisabled()).queue();
```

Comment:
Why does this not run the thanks/XP logic?

Reply:
it is run in the handleThanksCloseButton method

Reply:
I was specifically talking about the `else`-block.

Reply:
yeah, the logic is run in the handleThanksCloseButton method